packages/mdc-textfield/icon/foundation.ts

```diff
@@ -22,66 +22,59 @@
  */
 
 import {MDCFoundation} from '@material/base/foundation';
-import MDCTextFieldIconAdapter from './adapter';
+import {EventType} from '@material/base/index';
+import {MDCTextFieldIconAdapter} from './adapter';
 import {strings} from './constants';
 
+const CLICK_KEYDOWN_EVENTS: EventType[] = ['click', 'keydown'];
 
-/**
- * @extends {MDCFoundation<!MDCTextFieldIconAdapter>}
- * @final
- */
-class MDCTextFieldIconFoundation extends MDCFoundation {
-  /** @return enum {string} */
+class MDCTextFieldIconFoundation extends MDCFoundation<MDCTextFieldIconAdapter> {
   static get strings() {
     return strings;
   }
 
   /**
-   * {@see MDCTextFieldIconAdapter} for typing information on parameters and return
-   * types.
-   * @return {!MDCTextFieldIconAdapter}
+   * See {@link MDCTextFieldIconAdapter} for typing information on parameters and return types.
    */
-  static get defaultAdapter() {
-    return /** @type {!MDCTextFieldIconAdapter} */ ({
-      getAttr: () => {},
-      setAttr: () => {},
-      removeAttr: () => {},
-      setContent: () => {},
-      registerInteractionHandler: () => {},
-      deregisterInteractionHandler: () => {},
-      notifyIconAction: () => {},
-    });
+  static get defaultAdapter(): MDCTextFieldIconAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      getAttr: () => null,
+      setAttr: () => undefined,
+      removeAttr: () => undefined,
+      setContent: () => undefined,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      notifyIconAction: () => undefined,
+    };
+    // tslint:enable:object-literal-sort-keys
   }
 
-  /**
-   * @param {!MDCTextFieldIconAdapter} adapter
-   */
-  constructor(adapter) {
-    super(Object.assign(MDCTextFieldIconFoundation.defaultAdapter, adapter));
+  private savedTabIndex_: string | null;
+  private readonly interactionHandler_: EventListener;
```

Comment:
Should be `SpecificEventListener<'click'> | SpecificEventListener<'keydown'>`

Reply:
Done